types/express/package.json

```diff
@@ -1,13 +1,13 @@
 {
     "private": true,
     "name": "@types/express",
-    "version": "4.17.9999",
+    "version": "5.0.9999",
```

Comment:
Just so I'm clear, express v5 contains _no_ type changes? With all of the other stuff in this PR, I would have thought something would change here to warrant then, right?

Also, is express v5 actually out? Noticing that it's not marked as "latest". Typically we don't do prereleases, but it is an actual version number, so I assume that's okay?

(We may want to preserve v4 in a subdir if we think that it's going to be needing an update in the future, which I think is likely?)

Reply:
> Just so I'm clear, express v5 contains no type changes?

No other _type_ changes than properly supported async handlers.
There are changes that do not affect currently declared types.

> I would have thought something would change here to warrant then, right?

I'm not sure I understand the question.

> is express v5 actually out?

Yes, `5.0.0` is published, I'm using it myself in several projects.

> Noticing that it's not marked as "latest".

That's correct. It's published with dist-tag `next` at the moment.
This is because not everything is ready to invite broad audience to migrate to v5, in particular documentation AND the DT types which is exactly I'm helping to accomplish in this PR, @jakebailey 
https://github.com/expressjs/express/issues/5944#issuecomment-2341980442

> Typically we don't do prereleases, but it is an actual version number, so I assume that's okay?

It's not a pre-release, it's production ready release of the pure JS code community awaited for 10 years.
And I'm trying to make it usable and fully available for Typescript community.
https://github.com/expressjs/express/releases/tag/v5.0.0
https://www.npmjs.com/package/express/v/5.0.0

> We may want to preserve v4 in a subdir

I'm not sure it works and why, but I trust your expertise.

Reply:
> No other type changes than properly supported async handlers.

Right, but this PR doesn't contain any of those changes, no? I guess I was just expecting `types/express` to contain a diff that wasn't just the `package.json`...

> > We may want to preserve v4 in a subdir
> 
> I'm not sure how it works and why, but I trust your expertise.

More or less I mean to just switch back to main, save the `types/express` dir, switch back to this PR, move the saved dir to `types/express/v4`, then add `/v4/` to `.npmignore`.

Then, we can still maintain v4, if we're expecting v4 to continue to need maintenance. But, I'm not personally in a position to say if that's true.

Reply:
Oh, weird. I figured that one was in response to another change...

Reply:
> More or less I mean to just switch back to main, save the types/express dir, switch back to this PR, move the saved dir to types/express/v4, then add /v4/ to .npmignore.

Clear, thank you. Would you like me to do that in this PR? (I can)

> Then, we can still maintain v4, if we're expecting v4 to continue to need maintenance.

I think it' a good idea, it's safe and should cause no harm, @jakebailey

Reply:
Yeah, I would do that. You'll also need to do that for `express-serve-static-core`, which I didn't realize was a critical dep.

Reply:
(And change `express@4`'s dep on `express-serve-static-core` to `4` too)